packages/module-services/src/bookmarks/get/generate-endpoint.ts

```diff
@@ -0,0 +1,22 @@
+import { UnsupportedApiVersion } from '@equinor/fusion-framework-module-services/errors';
+import { ApiVersions, GetBookmarkArgs } from './types';
+
+/**
+ * Method for generating endpoint for getting bookmark by id
+ */
+export const generateEndpoint = <TVersion extends ApiVersions>(
+    version: TVersion,
+    args: GetBookmarkArgs<TVersion>
+) => {
+    switch (version) {
+        case 'v2':
+            throw new UnsupportedApiVersion(version);
+        case 'v1':
+        default: {
+            const { id } = args as { id: string };
+            const params = new URLSearchParams();
+            params.append('api-version', '1.0');
```

Comment:
since the mapping from enum is gone, you need to asses on default what the version will be. `version` could be `1.2-beta.1`.

Reply:
How will this assessment be done? is there a service providing current version if all endpoints or will this be done in code?

Reply:
https://github.com/equinor/fusion-framework/blob/604245a410116a9f055185d921dccf45cf5cf259/packages/module-services/src/context/get/generate-endpoint.ts#L13

if you look at the logic in existing code it will either fetch the version from version map or use the provided string.

in the proposed code, api version will always be `1.0` since there are no reference map of allowed version.

the idea of the map was:
```ts
enum Version {
'V1' = '1.2'
'V2' = '2.0'
}

const resolveVersion = (key: string | keyof typeof Version) => Version[key] ?? key;

// expect 1.2
const apiVersion = resolveVersion('V1');

// expect 2.1-beta.1
const apiVersion = resolveVersion('2.1-beta.1');
```

Reply:
the reason for this lingo is that a developer can easily use another api-version without rewriting the whole logic from scratch

Reply:
> @odinr How will this assessment be done? is there a service providing current version if all endpoints or will this be done in code?

developers should __NEVER__ lean on _current_ version, the version is explicit, and the developer __must__ have a relation to what version is consumed.